src/slice.rs

```diff
@@ -163,6 +183,7 @@ impl fmt::Display for SliceOrIndex {
                     write!(f, ";{}", step)?;
                 }
             }
+            AxisSliceInfo::NewAxis => write!(f, "NewAxis")?,
```

Comment:
stupid but new idea I had was to use `stringify!(NewAxis)` etc when doing this. It's like half a level cleaner code for refactoring.